src/pybind/mgr/dashboard/frontend/src/app/shared/services/feature-toggles.service.ts

```diff
@@ -15,11 +15,17 @@ export class FeatureTogglesService {
   readonly REFRESH_INTERVAL: number = 20000;
   private featureToggleMap$: FeatureTogglesMap$;
 
-  constructor(private http: HttpClient) {
-    this.featureToggleMap$ = timer(0, this.REFRESH_INTERVAL).pipe(
-      flatMap(() => this.http.get<FeatureTogglesMap>(this.API_URL)),
-      shareReplay(1)
-    );
+  constructor(private http: HttpClient, private zone: NgZone) {
+    this.featureToggleMap$ = this.http.get<FeatureTogglesMap>(this.API_URL).pipe(shareReplay(1));
+    this.zone.runOutsideAngular(() => {
+      window.setInterval(() => {
```

Comment:
Using RxJS ``timer`` does not need the use of ``NgZone``. Maybe you simply need to add ``delay(0)`` to get it working.

Reply:
I'm not sure if I understand it. The timer works fine inside the Angular zone, but I've changed it to run outside of the Angular zone to prevent Protractor from waiting endlessly, as a timer is a recurring event that won't be finished at some point.